libs/cardano-ledger-binary/src/Cardano/Ledger/Binary/Decoding/Decoder.hs

```diff
@@ -482,6 +485,13 @@ decodeList decodeValue =
     (decodeListWith decodeValue)
 {-# INLINE decodeList #-}
 
+decodeNonEmptyList :: Decoder s a -> Decoder s [a]
```

Comment:
I think it will be cooler if we also reflect this invariant in the type:
```suggestion
decodeNonEmptyList :: Decoder s a -> Decoder s (NonEmpty a)
```

You can also replace the `DecCBOR` instance for `NonEmpty` with this function.